lib/aiko/led.py

```diff
@@ -71,8 +72,35 @@ def apply_dim(color, dimmer=None):
 # TODO: Now fails when called by exec(), see lib/aiko/mqtt.py
 # return tuple([int(element * dimmer) for element in color])
 
-def fill(color):
+# Allow setting dim from code or MQTT
+def set_dim(dimmer):
+    global dim
+    dim = dimmer
+
+# this is used to turn the neopixels back on to default brightness
+def reset_dim():
+  parameter = configuration.main.parameter
+  set_dim(parameter("dim", configuration.led.settings))
+
+# Take from -0.9 to 0.9 (+-0.1 is more typical) and adjust diming
+# value. Make sure it stays within 0 to 1
+def change_dim(change):
+    global dim
+    dim += change
+    if dim<0: dim=0
+    if dim>1: dim=1
+
+def print_dim():
+    print("Dim: ", dim)
+
+def fill(color, write=True):
   np.fill(apply_dim((color[0], color[1], color[2])))
+  if write: np.write()
```

Comment:
Does this work without the conditional?
```suggestion
  np.write()
```

Reply:
fill was originally written not to refresh the pixels. I don't think this is a common use case, so I made refresh the default, but left the option to run with False so that pixels don't get refreshed.

Reply:
Oh, sorry, for some reason I looked below to the `def write` instead of above to the `write=True` argument :/